sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -450,45 +450,45 @@ object DataSourceStrategy {
 
   private def translateLeafNodeFilter(predicate: Expression): Option[Filter] = predicate match {
     case expressions.EqualTo(PushableColumn(name), Literal(v, t)) =>
-      Some(sources.EqualTo(name, convertToScala(v, t)))
+      Some(sources.EqualTo(name, convertToScala(v, t, false)))
```

Comment:
I guess we're treating this as a temp fix for Spark 3.0?
Looks like ideally we should support Java 8 datetime instances for this interface as well when `spark.sql.datetime.java8API.enabled` is enabled. It could cause more confusion. In addition, seems like `spark.sql.datetime.java8API.enabled` is disabled by default, too.

Reply:
I think it's problematic to let the java8 config also control the value type inside `Filter`, as it can break existing DS v1 implementations. It's a bit unfortunate that we don't document clearly what the value type can be for `Filter`, but if we do, it's not user-friendly to say "the value type depends on xxx config". This just makes it harder to implement data source filter pushdown.

Reply:
@HyukjinKwon Taking into account https://github.com/apache/spark/pull/23811#issuecomment-580443958, the flag won't be enabled by default in the near future.